snap.svg.zpd.js

```diff
@@ -403,50 +464,77 @@ SVGElement.prototype.getTransformToElement = SVGElement.prototype.getTransformTo
                     delta = event.detail / -9;       // Mozilla
                 }
 
-                var z = Math.pow(1 + zpdElement.options.zoomScale, delta);
+                _handleZoomingEvent(event, zpdElement, delta);
+            };
+            
+            var handleTouchMove = function handleTouchMove (event) {
+
+                if (!zpdElement.options.zoom || !zpdElement.options.touch) {
+                    return;
+                }
+
+                if (event.preventDefault) {
+                    event.preventDefault();
+                }
+
+                if (!snapsvgzpd.enable) return;
 
-                var g = zpdElement.element.node;
+                event.returnValue = false;
 
-                var p = _getEventPoint(event, zpdElement.data.svg);
+                // If multi-touch is true, then we are zooming instead of panning or dragging
+                if (zpdElement.data.touchZoom) {
 
-                p = p.matrixTransform(g.getCTM().inverse());
+                    var distance = _getMultiTouchDistance(event);
 
-                // Compute new scale matrix in current mouse position
-                var k = zpdElement.data.root.createSVGMatrix().translate(p.x, p.y).scale(z).translate(-p.x, -p.y);
+                    if (zpdElement.data.prevZoomDistance != null) {
+                        var delta = 0;
 
-                _setCTM(g, g.getCTM().multiply(k), zpdElement.options.zoomThreshold);
+                        if (zpdElement.data.prevZoomDistance > distance) delta = -0.15;
+                        else delta = 0.15;
```

Comment:
Do you mind add a comment here to explain why this magic numbers 0.15 or 0.111111.
Useful for future code reading 😃